skrub/_dataframe/_common.py

```diff
@@ -955,3 +956,9 @@ def _replace_pandas(col, old, new):
 @replace.specialize("polars", argument_type="Column")
 def _replace_polars(col, old, new):
     return col.replace(old, new)
+
+
+def with_columns(df, **new_columns):
+    columns = {col_name: col(df, col_name) for col_name in column_names(df)}
+    columns.update({n: make_column_like(df, c, n) for n, c in new_columns.items()})
```

Comment:
as we discussed IRL, we will want to add a separate test to check that `make_column_like` does what we want when the input is already a column

Reply:
checking also that the result name is n even if the input column has a different name (that is already the case in the main branch but there is no test to check that aspect)